tools/spectral/ipa/rulesets/functions/exceptionExtensionFormat.js

```diff
@@ -0,0 +1,31 @@
+const ERROR_MESSAGE = 'IPA exceptions must have a valid rule name and a reason.';
+const RULE_NAME_PREFIX = 'xgen-IPA-';
+const REASON_KEY = 'reason';
+
+// Note: This rule does not allow exceptions
+export default (input, _, { path }) => {
+  const exemptedRules = Object.keys(input);
+  const errors = [];
+
+  exemptedRules.forEach((ruleName) => {
+    const exception = input[ruleName];
+    if (!isValidException(ruleName, exception)) {
+      errors.push({
+        path: path.concat([ruleName]),
+        message: ERROR_MESSAGE,
+      });
+    }
+  });
+
+  return errors;
+};
+
+function isValidException(ruleName, exception) {
+  const exceptionObjectKeys = Object.keys(exception);
+  return (
+    ruleName.startsWith(RULE_NAME_PREFIX) &&
```

Comment:
Q: Should we check if the `ruleName` is among the rules defined?

Reply:
Yes, I think you would need another function which loads `ipa-spectral.yaml` to a Spectral object and fetch the rule names from there

Reply:
I think it is fine by now, it can be an improvement for later